spec/enumerables_spec.rb

```diff
@@ -0,0 +1,139 @@
+require './script'
+require 'rspec'
+
+RSpec.describe Enumerable do
+  let(:arr) { [6, 4, 3, 5] }
+  let(:range) { (1..8) }
+  let(:string) { %w[red green factor] }
+  let(:block) { proc { |num| num < 4 } }
+
+  describe '#my_each' do
+    context 'when no block is given' do
+      it { is_expected.to be_truthy }
+    end
+
+    it 'returns the original array' do
+      expect(string.my_each { |x| x }).to eq(string)
+    end
+
+    it 'prints result to stdout from block calculation' do
+      expect { arr.my_each { |i| print i + 4 } }.to output(/10879/).to_stdout
+    end
+
+    it 'prints result to stdout with block given' do
+      expect { string.my_each { |i| print i.length } }.to output(/356/).to_stdout
+    end
+  end
+
+  describe '#my_each_with_index' do
+    context 'when no block is given' do
+      it { is_expected.to be_truthy }
+    end
+
+    it 'prints the element and corresponding index to stdout' do
+      expect { arr.my_each_with_index { |x, i| print x, i } }.to output(/60413253/).to_stdout
+    end
+
+    it 'returns original array when no output' do
+      expect(arr.my_each_with_index { block }).to eq(arr)
+    end
+  end
+
+  describe '#my_select' do
+    context 'when no block is given' do
+      it { is_expected.to be_truthy }
+    end
+
+    it 'returns even numbers from array of numbers' do
+      expect(arr.my_select(&:even?)).to eq([6, 4])
+    end
+  end
+
+  describe '#my_all?' do
+    it 'should return true' do
+      expect(arr.my_all?(&:odd?)).to be_falsy
+    end
+
+    it 'should return false' do
+      expect(arr.my_all? { |n| n < 10 }).to be_truthy
+    end
+  end
```

Comment:
Your specs should cover for negative and positive scenarios. For example in the above, 'should return true when the values in the arr are all odd'. There should be another test, 'should return false when the values in the arr are not all odd'.
For the test 'it should return false when all the values are less than 10', there should be a test for 'should return true when not all values in the arr are less than 10.'

Please do this for all your other tests. See [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/commonsense.png) from pathwright.